lib/Coffer/Util.hs

```diff
@@ -18,3 +24,26 @@ import Polysemy.Error ( runError, Error )
 catchAndReturn :: forall e r. Sem (Error e ': r) e -> Sem r e
 catchAndReturn action =
   either id id <$> runError @e action
+
+
+----------------------------------------------------------------------------
+-- Tomland helpers
+----------------------------------------------------------------------------
+
+{-# INLINE didimatch #-}
+didimatch
+  :: (b -> Either Text a)  -- ^ Mapper for consumer
+  -> (a -> Either Text b)  -- ^ Mapper for producer
+  -> TomlCodec a  -- ^ Source 'Codec' object
+  -> TomlCodec b  -- ^ Target 'Codec' object
+didimatch matchB matchA codec = Toml.Codec
+    { Toml.codecRead = \t -> case Toml.codecRead codec t of
+        Success a ->
+          case matchA a of
+            Left err -> Failure [Toml.ParseError $ Toml.TomlParseError err]
+            Right b -> Success b
+        Failure b -> Failure b
+    , Toml.codecWrite = \c -> case matchB c of
+        Left _ -> empty
+        Right d  -> Toml.codecWrite codec d >>= Toml.eitherToTomlState . matchA
```

Comment:
We shouldn't be swallowing the error in `Left`. How about this?

```suggestion
    , Toml.codecWrite = \b -> do
        a <- Toml.eitherToTomlState $ matchB b
        a' <- Toml.codecWrite codec a
        Toml.eitherToTomlState $ matchA a'
```

Reply:
Done

Reply:
Looks good!

As a sidenote: turns out the [implementation](https://hackage.haskell.org/package/tomland-1.3.3.1/docs/src/Toml.Codec.Types.html#eitherToTomlState) of `eitherToTomlState` _also_ swallows the error in the `Left` :weary: 

```hs
eitherToTomlState :: Either e a -> TomlState a
eitherToTomlState e = TomlState (either (const Nothing) Just e,)
```

Oh well, seems like the way `TomlState` is represented there's just no way of properly reporting errors in `codecWrite`...